docs/MostUsedConditions.md

```diff
@@ -0,0 +1,259 @@
+```php
+<?php
+$a = 14;
+if (13 === $a || 14 === $a) {
+    // do something
+}
+
+if (!(13 === $a && 15 === $a)) {
+    // do something else
+}
+```
+The SplitLogicalOperator-Processor will split the first if to `13 === $a` and `14 === $a`.
+The second if will be split to `!13 === $a` and `!15 === $a`.
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=SplitLogicalOperator --include-files=splitlogicaloperatorprocessor.php`
+
+<img src="./images/MostUsedConditions/splitlogicaloperatorprocessor.png" height="200">
+
+#### NegateBooleanNot
+This processor will negate all comparison operators when it's negated from outside. See [this](examples/MostUsedConditions/negatebooleannotprocessor.php) source-code:
+```php
+<?php
+$a = 14;
+if (!($a === 13)) {
+    // do something
+}
+
+if (!($a !== 14)) {
+    // do something else
+}
+
+if (!($a < 13)) {
+    // do something else
+}
+```
+The NegateBooleanNot-Processor will change the first if to `$a !== 13`, the second to `$a === 14` and the last to `$a > 13`. This processor is very nice in combination with [SplitLogicalOperator](#splitlogicaloperator) from above, but make sure to add this processor in the list after the SplitLogicalOperator
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=NegateBooleanNot --include-files=negatebooleannotprocessor.php`
+
+<img src="./images/MostUsedConditions/negatebooleannotprocessor.png" height="240">
+
+#### RemoveAssignment
+This processor will remove assignments in your conditions. See [this](examples/MostUsedConditions/removeassignmentprocessor.php) source-code:
+```php
+<?php
+if ($this->getUser()) {
+    // do something
+}
+
+if ($user = $this->getUser()) {
+    // do something else
+}
+```
+You want to know that the condition from $this->getUser() is used twice, but normally it would be counted as one plus one. This processor will remove `$user = ` from the second condition and will count 2 for `$this->getUser()`.
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=RemoveAssignment --include-files=removeassignmentprocessor.php`
+
+<img src="./images/MostUsedConditions/removeassignmentprocessor.png" height="200">
+
+#### RemoveDuplicateBooleanNot
+This processor removes duplicate negations. So if you have an complex condition, and split it off, it could be possible in the printed condition is something like `!!$user`. You can achieve this with [this](examples/MostUsedConditions/removeduplicatebooleannotprocessor.php) source-code and the processors [SplitLogicalOperator](#splitlogicaloperator) and [NegateBooleanNot](#negatebooleannot) together:
```

Comment:
```suggestion
This processor removes duplicate negations. So if you have a complex condition, and split it off, it could be possible for the printed condition to be something like `!!$user`. You can achieve this with [this](examples/MostUsedConditions/removeduplicatebooleannotprocessor.php) source-code and the processors [SplitLogicalOperator](#splitlogicaloperator) and [NegateBooleanNot](#negatebooleannot) together:
```